content/en/blog/_posts/2023-12-14-disabling-in-tree-cloud-provider-goes-beta.md

```diff
@@ -0,0 +1,210 @@
+you **are** on Azure, GCE, or vSphere, or you are upgrading from a version
+older than 1.28, then read on.
+
+Historically, Kubernetes has included code for a set of cloud providers that
+included AWS, Azure, GCE, OpenStack, and vSphere. Since the inception of
+[KEP-2395][kep2395] the community has been moving towards removal of that
+cloud provider code. The OpenStack provider code was removed in version 1.26,
+and the AWS provider code was removed in version 1.27. This means that users
+who are upgradging from one of the affected cloud providers and versions will
+need to modify their deployments.
+
+### Upgrading on Azure, GCE, or vSphere
+
+There are two options for upgrading in this configuration: migrate to external
+cloud controller managers, or continue using the in-tree provider code.
+Although migrating to external cloud controller managers is recommended,
+there are scenarios where continuing with the current behavior is desired.
+Please choose the best option for your needs.
+
+#### Migrate to external cloud controller managers
+
+Migrating to use external cloud controller managers is the recommended upgrade
+path as it aligns with the future of Kubernetes. To do this you will need to
+enable the `--cloud-provider=external` command line flag for the
+`kube-apiserver`, `kube-controller-manager`, and `kubelet` components. In
+addition you will need to deploy a cloud controller manager for your provider.
+
+Installing and running cloud controller managers is a larger topic than this
+post can address, if you would like more information on this process please
+read the documentation for [Cloud Controller Manager Administration][ccmadmin]
+and [Migrate Replicated Control Plane To Use Cloud Controller Manager][ccmha].
+See [below](#cloud-provider-integrations) for links to specific cloud provider
+implementations.
+
+#### Continue using in-tree provider code
+
+If you wish to continue using Kubernetes with the in-tree cloud provider code,
+you will need to modify the command line parameters for `kube-apiserver`,
+`kube-controller-manager`, and `kubelet` to disable the feature gates for
+`DisableCloudProviders` and `DisableKubeletCloudCredentialProviders`. To do
+this, add the following command line flag to the arguments for the previously
+listed commands:
+
+```
+--feature-gates=DisableCloudProviders=false,DisableKubeletCloudCredentialProviders=false
+```
+
+_Please note that if you have other feature gate modifications on the command
+line, they will need to include these 2 feature gates._
+
+**WARNING**: These feature gates will be locked to `true` in an upcoming
+release. Setting these feature gates to `false` should be used as a last
+resort, it is highly recommened to migrate to an external cloud controller
+manager.
+
+### Upgrading on other providers
+
+For providers other than Azure, GCE, or vSphere, good news, the external cloud
+controller manager should be already be in use. The code for AWS and OpenStack
```

Comment:
```suggestion
controller manager should be already be in use. The code for AWS and OpenStack
```

If I were the end user in this situation, "should be" would make me nervous. How would I tell if "should be" means "is" for my cluster?

Reply:
added an extra sentence here to give some action on how to check.